consts/src/lib.rs

```diff
@@ -172,6 +187,8 @@ impl Default for EdhocMessageBuffer {
 
 pub trait MessageBufferTrait {
     fn new() -> Self;
+    fn get(self, index: usize) -> Result<u8, AccessError>;
+    fn get_slice<'a>(&'a self, start: usize, len: usize) -> Result<&'a [u8], AccessError>;
```

Comment:
The standard library uses a single [get function](https://doc.rust-lang.org/std/primitive.slice.html#method.get) that utilizes the SliceIndex trait. Could we do this here as well?

(Those two items combined might pave the way toward plugging in more widespread buffer implementations for production builds, and/or working with anything that is DerefMut<[u8]>).